samples/bluetooth/peripheral_hids_keyboard/src/main.c

```diff
@@ -0,0 +1,629 @@
+ * Here, '-' refers to release, 'c' refers to the key character being
+ * transmitted.
+ * Therefore 7 notifications will be sent.
+ *
+ * @param string    Key codes to be sent.
+ * @param len       Number of buttons to be sent.
+ * @param conn      Connection handler.
+ * @param boot_mode Information if boot mode protocol is selected.
+ *
+ * @return 0 on success or negative error code.
+ */
+int key_press_conn_send(u8_t *string,
+			u8_t len,
+			struct bt_conn *conn,
+			bool boot_mode)
+{
+	int err = 0;
+	u8_t  offset = 0;
+	u8_t  data[INPUT_REPORT_KEYS_MAX_LEN];
+	u32_t shift_btn;
+
+	while (offset <= len) {
+		memset(data, 0, sizeof(data));
+		memcpy(data + SCAN_CODE_POS + offset, string + offset,
+		       len - offset);
+
+		gpio_pin_read(gpio_devs[SHIFT_BUTTON_PORT], SHIFT_BUTTON,
+			      &shift_btn);
+		if (shift_btn == SHIFT_BUTTON_PRESSED) {
+			data[MODIFIER_KEY_POS] |= SHIFT_KEY_CODE;
+		}
+
+		if (boot_mode) {
+			err = hids_boot_kb_inp_rep_send(&hids_obj, conn,
+							data, sizeof(data),
+							NULL);
+		} else {
+			err = hids_inp_rep_send(&hids_obj, conn,
+						INPUT_REP_KEYS_REF_ID,
+						data, sizeof(data),
+						NULL);
+		}
+		if (err) {
+			break;
+		}
+		offset++;
+	}
+	return err;
+}
+
+
+void key_press_send(u8_t *string, u8_t len)
+{
+	if (len > KEYS_MAX_LEN) {
+		printk("Text is too big\n");
+		return;
+	}
+
+	for (size_t i = 0; i < CONFIG_BT_GATT_HIDS_MAX_CLIENT_COUNT; i++) {
+		if (conn_mode[i].conn) {
+			(void)key_press_conn_send(string, len,
+						  conn_mode[i].conn,
+						  conn_mode[i].in_boot_mode);
+		}
+	}
+}
+
+
+void button_pressed(struct device *gpio_dev, struct gpio_callback *cb,
+		    u32_t pins)
+{
+	static u8_t *chr = hello_world_str;
+
+	if (pins & (1 << SW0_GPIO_PIN)) {
+		key_pack_queue(chr++, 1);
+		if (chr == (hello_world_str + sizeof(hello_world_str))) {
+			chr = hello_world_str;
+		}
+	}
+}
+
+
+void configure_gpio(void)
+{
+	static const u32_t pin_id[4] = { SW0_GPIO_PIN,
+					 SW1_GPIO_PIN,
+					 SW2_GPIO_PIN,
+					 SW3_GPIO_PIN };
+	static const char *port_name[4] = { SW0_GPIO_CONTROLLER,
+					    SW1_GPIO_CONTROLLER,
+					    SW2_GPIO_CONTROLLER,
+					    SW3_GPIO_CONTROLLER };
+
+	for (size_t i = 0; i < ARRAY_SIZE(pin_id); i++) {
+		gpio_devs[i] = device_get_binding(port_name[i]);
+		if (gpio_devs[i]) {
+			printk("%s(): port %zu bound\n", __func__, i);
+
+			gpio_pin_configure(gpio_devs[i], pin_id[i],
+					   GPIO_PUD_PULL_UP | GPIO_DIR_IN |
+					   GPIO_INT | GPIO_INT_EDGE |
+					   GPIO_INT_ACTIVE_LOW);
+			gpio_init_callback(&gpio_cbs[i], button_pressed,
+					   BIT(pin_id[i]));
+			gpio_add_callback(gpio_devs[i], &gpio_cbs[i]);
+			gpio_pin_enable_callback(gpio_devs[i], pin_id[i]);
+		}
+	}
+
+	led_dev = device_get_binding(LED0_GPIO_CONTROLLER);
+
+	gpio_pin_configure(led_dev, CAPS_LOCK_LED, GPIO_DIR_OUT);
+	gpio_pin_write(led_dev, CAPS_LOCK_LED, 1);
+}
+
+
+void main(void)
+{
+	int err;
+	struct key_pack keys;
+
+	printk("Start zephyr\n");
```

Comment:
Why not print the name of the project instead? :)

Reply:
Done